### PR TITLE
Align formatting of APM rules with other rules docs

### LINKED ISSUE #4040: [Request]: Add list of valid action variables to APM alerting topics
### Description

Most of the topics about creating alerting rules for observability provide a list of action variables that are valid in that rule. The APM rules documented ([documented under APM in the Kibana Guide](https://www.elastic.co/guide/en/kibana/8.13/apm-alerts.html)) should, but do not. We are documenting valid action variables for each rule because this was a specific request by users who wanted to to see this information in the docs.

This is related to the work done in https://github.com/elastic/observability-docs/pull/3693).

NOTE: I started to update the content in place (where it lives in the Kibana Guide), but that won't work because the list of valid action variables is not the same across all the APM-related rules. I think it would be better to wait until after the rules are restructured before trying to make these changes. 

### Resources

n/a

### Which documentation set does this change impact?

Stateful and Serverless

### Feature differences

n/a

### What release is this request related to?

N/A

### Collaboration model

The documentation team

### Point of contact.

**Main contact:** TBD

**Stakeholders:**

### PR DESCRIPTION
**Pull request by colleenmcginnis**
## Description

Aligns formatting of APM rules with other rules docs. This includes:

* Splitting the current APM rules docs from one long page, to a separate page for each rule.
* Adding lists of filters, conditions, action types, and valid action variables.
* Adding an intro to APM rules in the APM section of the Observability guide with links to relevant rules docs.

~~So far, I've fully written the [APM Anomaly rule](https://observability-docs_bk_4366.docs-preview.app.elstc.co/guide/en/observability/master/apm-anomaly-rule.html) page. If the team thinks this is a good approach, I'll finish building out the content on the pages for the rest of the APM-related rules.~~

Preview: https://observability-docs_bk_4366.docs-preview.app.elstc.co/guide/en/observability/master/apm-alerts.html

### Documentation sets edited in this PR

_Check all that apply._

- [x] Stateful (`docs/en/observability/*`)
- [ ] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue

Closes #4040

## Checklist

- [x] ~~Product/Engineering Review~~
- [x] Writer Review

### Follow-up tasks

_Select one._

* This PR does _not_ need to be ported to another doc set because:
  - [ ] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [ ] The PR contains edits to both doc sets (serverless _and_ stateful)
  - [x] Other:
    I'm not going to port to serverless right now because: 
    * In the serverless docs, the APM rules are already in the alerting section.
    * There are some different options for serverless rules vs stateful rules.
    * We can align on the formatting when addressing https://github.com/elastic/observability-docs/issues/4128.
    Let me know if you have any concerns about this approach!
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>

### CODE REVIEW

**Comment by colleenmcginnis** [comment]
>So far, I've fully written the [APM Anomaly rule](https://observability-docs_bk_4366.docs-preview.app.elstc.co/guide/en/observability/master/apm-anomaly-rule.html) page. If the team thinks this is a good approach, I'll finish building out the content on the pages for the rest of the APM-related rules.
>
>Preview: https://observability-docs_bk_4366.docs-preview.app.elstc.co/guide/en/observability/master/apm-alerts.html

@bmorelli25 and/or @dedemorton can you take a look and let me know what you think?

File: docs/en/observability/apm-error-count-threshold-rule.asciidoc [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+Alert when the number of errors in a service exceeds a defined threshold. Error count rules can be set at the
+environment level, service level, and error group level.
+
+[discrete]
+[[apm-error-count-threshold-rule-filters-conditions]]
+== Filters and conditions
+
+// TO DO: Should we explain what an error grouping key is?
+// I couldn't find a good explanation anywhere else...
```
Comment (bmorelli25) [inline docs/en/observability/apm-error-count-threshold-rule.asciidoc:12]: Yes. You can find information on how the error grouping key is generated in this thread: https://discuss.elastic.co/t/about-the-principle-of-error-grouping-key/283622

Note that the file Andrew refers to in his answer now lives here: https://github.com/elastic/apm-data/blob/main/model/modelprocessor/groupingkey.go.

File: docs/en/observability/apm-error-count-threshold-rule.asciidoc [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+Supported built-in integrations include:
+
+include::../shared/alerting-and-rules/alerting-connectors.asciidoc[]
+
+[discrete]
+=== Action frequency
+
+include::../shared/alerting-and-rules/generic-apm-action-frequency.asciidoc[]
+
+// TO DO: add image?
```
Comment (bmorelli25) [inline docs/en/observability/apm-error-count-threshold-rule.asciidoc:69]: I'm not sold on adding an image here.

**Comment by bmorelli25** [comment]
Good stuff! I like where this is heading.

**Review by bmorelli25 — APPROVED** [review]
🌸 ❗❗

**Pull request merged**